### PR TITLE
Use default timezone for logs if logtimezone unset

### LINKED ISSUE #8212: Feature Request: add 'logtimezone' automatically to config.php during install
It would be sensful if during the installation / finalisation of owncloud, the timezone of the server is added to config php automatically.
Logs created by oC would then be with the right timzone from beginning.

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
Augmentation of \OC\DateTimeZone with the `getSystemTimeZone()` method, which returns the system timezone (`date_default_timezone_get()`). This is then used if `logtimezone` is unset for the log timezone, instead of the default of UTC.

This does mean however that the setting of UTC as the default timezone in base.php has been removed. I don't know the implications of this, but hopefully our code is robust enough to cope. (that line was introduced in December 2012 in 7d811e57e6c7942964d6937a8cc1113839e25968)

Fixes #8212

cc @PVince81 @DeepDiver1975 @LukasReschke @MorrisJobke

### CODE REVIEW

**Comment by karlitschek** [comment]
This need to be tested _very_ well. I thunk the right approach is to always use UTC internally and only use user configurable or auto detect timezones for output like logfiles and UI

**Comment by Xenopathic** [comment]
@karlitschek What could a non-UTC timezone affect? We use POSIX timestamps everywhere AFAIK, and then convert that into a real time with timezone according to the user or server's preference.

**Comment by karlitschek** [comment]
@Xenopathic I think you are right. But we still need to test this properly. There is the possibility of breakage in special corner-cases.

**Comment by DeepDiver1975** [comment]
:-1: a server NEVER runs with some special time zone config what so ever! The server all by it self is a neutral entity moving data from left to right. Generally speaking it is always the client's responsibility to properly perform timezone transformations.

File: tests/lib/datetimezone.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+namespace Test;
+
+class DateTimeZone extends TestCase {
+	protected $config;
+	protected $session;
+
+	protected function setUp() {
+		date_default_timezone_set('UTC');
```
Comment (nickvergessen) [inline tests/lib/datetimezone.php]: please reset to original value after the test?

**Comment by PVince81** [comment]
I agree with @DeepDiver1975. The server must always run in UTC mode and the view (the UI) must do the necessary transformations to display times in the viewer's timezone.
This also means that all dates need to be stored in UTC too in the database.

@Xenopathic unless we misunderstood what this PR is about ?

**Comment by Xenopathic** [comment]
@PVince81 This PR won't affect the times stored in the database or what the user sees - we store times as a timestamp and convert it to the user's timezone (conviniently, in `DateTimeZone->getTimeZone()`). This just changes the default timezone used in a couple of places, such as the default user timezone (it usually makes sense for a user's default timezone to match the server, until they change it) and default timezones used elsewhere, like in the logs (can be overridden with `logtimezone`).

I think @karlitschek's comment summarised the problems quite well: this *shouldn't* affect our existing code, but it might, so I no longer think this is the right approach. There are cases where getting the server timezone is useful, but `date_default_timezone_get()` should always return UTC.

**Comment by Xenopathic** [comment]
Note: please don't delete the branch yet, I'll reopen this when I've made the necessary changes
